openmc/deplete/chain.py

```diff
@@ -451,3 +452,168 @@ def form_matrix(self, rates):
+        no_capture = set()
+
+        # Check for validity before manipulation
+
+        for parent, sub in branch_ratios.items():
+            if parent not in self:
+                if strict:
+                    raise KeyError(parent)
+                missing_parents.add(parent)
+                continue
+
+            # Make sure all products are present in the chain
+
+            prod_flag = False
+
+            for product in sub:
+                if product not in self.nuclide_dict:
+                    if strict:
+                        raise KeyError(product)
+                    missing_products[parent] = product
+                    prod_flag = True
+                    break
+
+            if prod_flag:
+                continue
+
+            # Make sure this nuclide has capture reactions
+
+            indexes = []
+            for ix, rx in enumerate(self[parent].reactions):
+                if rx.type == "(n,gamma)":
+                    indexes.append(ix)
+                    if "_m" not in rx.target:
+                        grounds[parent] = rx.target
+
+            if len(indexes) == 0:
+                if strict:
+                    raise AttributeError(
+                        "Nuclide {} does not have capture reactions in "
+                        "this {}".format(parent, self.__class__.__name__))
+                no_capture.add(parent)
+                continue
+
+            capt_ix_map[parent] = indexes
+
+            this_sum = sum(sub.values())
+            check_less_than(parent + " ratios", this_sum, 1.0, True)
+            sums[parent] = this_sum
+
+        if len(missing_parents) > 0:
+            print("The following nuclides were not found in {}: {}".format(
+                self.__class__.__name__, ", ".join(sorted(missing_parents))))
+
+        if len(no_capture) > 0:
+            print("The following nuclides did not have capture reactions: "
+                  "{}".format(", ".join(sorted(no_capture))))
+
+        if len(missing_products) > 0:
+            tail = ("{} -> {}".format(k, v)
+                    for k, v in sorted(missing_products.items()))
+            print("The following products were not found in the {} and "
+                  "parents were unmodified: \n{}".format(
+                      self.__class__.__name__, ", ".join(tail)))
+
+        # Insert new ReactionTuples with updated branch ratios
+
+        for parent_name, capt_index in capt_ix_map.items():
+
+            parent = self[parent_name]
+            new_ratios = branch_ratios[parent_name]
+            capt_index = capt_ix_map[parent_name]
+
+            # Assume Q value is independent of target state
+            capt_Q = parent.reactions[capt_index[0]].Q
+
+            # Remove existing capture reactions
+
+            for ix in reversed(capt_index):
+                parent.reactions.pop(ix)
+
+            all_meta = False
+
+            for tgt, br in new_ratios.items():
+                all_meta |= ("_m" in tgt)
```

Comment:
Wouldn't `all_meta` be true even if only one of these evaluated as true? Seems like you'd want:
```Python
all_meta = True
for tgt, br in new_ratios.items():
    all_meta = all_meta and ("_m" in tgt)
    ...
```
Also, note that `|` is a bitwise-or operator and happens to work because booleans can be treated as integers (e.g., `True + 2` is a valid expression). However, to state the intent of a logical or/and, you should be using the `or` or `and` operators.